### PR TITLE
fix: misc. issues with closing

### PR DESCRIPTION
**Pull request by 0xArdi**
This PR fixes multiple issues introduced with `market_closing`.

### CODE REVIEW

**Review by Adamantios — APPROVED** [review]

File: packages/valory/skills/market_creation_manager_abci/rounds.py [hunk 1]
```diff
@@ -704,12 +704,29 @@ class CloseMarketsRound(CollectSameUntilThresholdRound):
...
+
         if self.threshold_reached:
             if self.most_voted_payload == self.ERROR_PAYLOAD:
                 return self.synchronized_data, Event.ERROR
             if self.most_voted_payload == self.NO_TX:
                 return self.synchronized_data, Event.NO_TX
-            state = self.synchronized_data.update(
+            synced_data = cast(SynchronizedData, self.synchronized_data)
+
+            # Fix to ensure properties are present on the SynchronizedData
```
Comment (jmoreira-valory) [inline packages/valory/skills/market_creation_manager_abci/rounds.py:715]: Isn't this already on main?

File: packages/valory/skills/market_creation_manager_abci/rounds.py [hunk 2]
```diff
@@ -746,7 +763,11 @@ class MarketCreationManagerAbciApp(AbciApp[Event]):
     """MarketCreationManagerAbciApp"""
 
     initial_round_cls: AppState = CollectRandomnessRound
-    initial_states: Set[AppState] = {CollectRandomnessRound, PostTransactionRound}
+    initial_states: Set[AppState] = {
+        CollectRandomnessRound,
+        PostTransactionRound,
+        CloseMarketsRound,
```
Comment (jmoreira-valory) [inline packages/valory/skills/market_creation_manager_abci/rounds.py:769]: Why?

File: packages/valory/skills/market_creation_manager_abci/rounds.py (hunk 1 above)
Comment (0xArdi) [reply packages/valory/skills/market_creation_manager_abci/rounds.py:715]: not on this state

File: packages/valory/skills/market_creation_manager_abci/rounds.py (hunk 2 above)
Comment (0xArdi) [reply packages/valory/skills/market_creation_manager_abci/rounds.py:769]: the way it was previously wired, CloseMarkets can end up not being called. Now it is the first behaviour that is called after registration, in order to do that it must be a initial state.

**Review by Adamantios — APPROVED** [review]

**Review by jmoreira-valory — APPROVED** [review]

**Pull request merged**